docs/reference/vectors/vector-functions.asciidoc

```diff
@@ -0,0 +1,283 @@
+<2> To take advantage of the script optimizations, provide a query vector as a script parameter.
+
+NOTE: If a document's dense vector field has a number of dimensions
+different from the query's vector, an error will be thrown.
+
+Similarly, for sparse_vector fields, `cosineSimilaritySparse` calculates cosine similarity
+between a given query vector and document vectors.
+
+[source,js]
+--------------------------------------------------
+GET my_index/_search
+{
+  "query": {
+    "script_score": {
+      "query": {
+        "match_all": {}
+      },
+      "script": {
+        "source": "cosineSimilaritySparse(params.query_vector, doc['my_sparse_vector']) + 1.0",
+        "params": {
+          "query_vector": {"2": 0.5, "10" : 111.3, "50": -1.3, "113": 14.8, "4545": 156.0}
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+For dense_vector fields, `dotProduct` calculates the measure of
+dot product between a given query vector and document vectors.
+
+[source,js]
+--------------------------------------------------
+GET my_index/_search
+{
+  "query": {
+    "script_score": {
+      "query": {
+        "match_all": {}
+      },
+      "script": {
+        "source": """
+          double value = dotProduct(params.query_vector, doc['my_dense_vector']);
+          return sigmoid(1, Math.E, -value); <1>
+        """,
+        "params": {
+          "query_vector": [4, 3.4, -0.2]
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+<1> Using the standard sigmoid function prevents scores from being negative.
+
+Similarly, for sparse_vector fields, `dotProductSparse` calculates dot product
+between a given query vector and document vectors.
+
+[source,js]
+--------------------------------------------------
+GET my_index/_search
+{
+  "query": {
+    "script_score": {
+      "query": {
+        "match_all": {}
+      },
+      "script": {
+        "source": """
+          double value = dotProductSparse(params.query_vector, doc['my_sparse_vector']);
+          return sigmoid(1, Math.E, -value);
+        """,
+         "params": {
+          "query_vector": {"2": 0.5, "10" : 111.3, "50": -1.3, "113": 14.8, "4545": 156.0}
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+For dense_vector fields, `l1norm` calculates L^1^ distance
+(Manhattan distance) between a given query vector and
+document vectors.
+
+[source,js]
+--------------------------------------------------
+GET my_index/_search
+{
+  "query": {
+    "script_score": {
+      "query": {
+        "match_all": {}
+      },
+      "script": {
+        "source": "l1norm(params.queryVector, doc['my_dense_vector'])",
+        "params": {
+          "queryVector": [4, 3.4, -0.2]
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+NOTE: Unlike `cosineSimilarity` that represent similarity, `l1norm` and
```

Comment:
I had a couple suggestions around this transformation:
- We could include a 1 in the denominator (`1/ l1norm(params.queryVector, doc['my_dense_vector'])`) to avoid division by 0 when a document vector matches the query exactly.
- Would it make sense to use this transformation in the search themselves, instead of just including it in a note? We could give a description of it in a callout, as we do for `cosineSimilarity` and `dotProduct`. It would be harder to miss that way, and I also think it makes the examples more helpful + realistic, since it seems very rare to score by distance descending.

Reply:
@jtibshirani Thanks these are very good suggestions, I have addressed them in the last commit

Reply:
Thanks @mayya-sharipova, the new commit you added looks good to me!